pkg/skaffold/runner/runner.go

```diff
@@ -228,13 +231,33 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*la
 				logger.Unmute()
 			}
 		}()
+		for _, a := range changed.dirtyArtifacts {
+			s, err := sync.NewItem(a.artifact, a.events, r.builds)
+			if err != nil {
+				return errors.Wrap(err, "sync")
+			}
+			if s != nil {
+				changed.AddResync(s)
+			}
+			if s == nil {
+				changed.AddRebuild(a.artifact)
+			}
+		}
 
 		switch {
 		case changed.needsReload:
 			logger.Stop()
 			return ErrorConfigurationChanged
-		case len(changed.dirtyArtifacts) > 0:
-			bRes, err := r.Build(ctx, out, r.Tagger, changed.dirtyArtifacts)
+		case len(changed.needsResync) > 0:
+			for _, s := range changed.needsResync {
+				if err := r.Syncer.Sync(s); err != nil {
+					logrus.Warnln("Skipping build and deploy due to sync error:", err)
+					return nil
+				}
+				color.Default.Fprintf(out, "Synced files for %s...\nCopied: %s\nDeleted: %s\n", s.Image, s.Copy, s.Delete)
```

Comment:
Maybe print the number of files in info and the names in debug

Reply:
Should there be any message in stdout?

stdout: ?
info: # of files synced
debug: filenames

Reply:
I like info: # of files synced
debug: filenames

Reply:
done